manager/state/raft/util.go

```diff
@@ -26,6 +26,14 @@ func dial(addr string, protocol string, creds credentials.TransportAuthenticator
 	return grpc.Dial(addr, grpcOptions...)
 }
 
+// handshake pings a node back using its advertised address to check if it is
+// reachable or not.
+func handshake(addr string, creds credentials.TransportAuthenticator) error {
+	client, err := dial(addr, "tcp", creds, 3*time.Second)
```

Comment:
Don't do that. It doesn't work very well. `grpc.Dial` is not a dialer. Treat it as `grpc.NewConnectionPool`.

What happened to the ping service we talked about?